newIDE/app/src/SceneEditor/ObjectsAdditionalService.js

```diff
@@ -0,0 +1,118 @@
+// @flow
```

Comment:
We need to find a better name for this I guess, right now it could handle objects and instances, "Objects"AdditionalService seems that it is only related to objects. This could become quite generic and handle much more "events" (not the most accurate word) in the future, not just "onObjectAdded" and "onInstanceAdded", therefore we could potentially automate a lot of stuff. :)

Reply:
I would recommend moving this into `newIDE/app/src/Hints`. `Hints/index.js` is already used to expose to the IDE some "hints" to display to the user for some object/behaviors.

Probably we could also simply this because I think the "service" is a bit overengineered.
Basically, we just need two functions `onObjectAdded` and `onInstanceAdded` and they can fetch the object type (in case of instance) by themselves. Internally, they can use a map to associate the object type to a closure called to do the work, but that's just an implementation detail.

Reply:
In some way I think of this extra work as a "boosted hint": it's a hint so performant that we're doing work for the user :) hence why I was talking about this `Hints` folder.

Reply:
Should I move the code in `Hints/index.js` then? And have an object which has two properties (functions), `onObjectAdded` and `onInstance` added, which then does whatever it needs to do (map to correct object types, do extra work, send back infoBarDetails, etc).

Reply:
Maybe in the Hints folder but still in a file with a different name, like "ObjectsAdditionalWork.js" (not super fan of "Work" but well!).
It's fine to have "Object" in the name as instances are very tied to the concept of object. 

I was thinking of alternate names like "ObjectAutomatedWork" but can't find anything really satisfying!